.github/workflows/python-client.yml

```diff
@@ -0,0 +1,51 @@
+# .github/workflows/python-client.yml
+name: python-client
+env:
+  WORKDIR: python-client
+
+on: 
+  push: 
+    paths: 
+      - 'python-client/**'
+    branches:
+      - main
+  pull_request: 
+    paths: 
+      - 'python-client/**'
+
+jobs:
+  test-and-deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      # Setup Python (faster than using Python container)
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.6"
+
+      - name: Install pinned dependencies
+        run: |
+          pip install -r $WORKDIR/requirements.txt
+
+      - name: Install
+        run: |
+          pip install -e $WORKDIR
+
+      - name: Test Annotate Plugin
+        run: pip install pytest-github-actions-annotate-failures
+
+      - name: Test
+        run: |
+          pytest $WORKDIR
+        
+      - name: Publish distribution 📦 to PyPI
+        if: github.ref == 'refs/heads/main'
```

Comment:
this checks 'if we're in main, for a merge/etc', but doesn't check the directory being changed.
Maybe this all only runs if the python parts are changing though?

Reply:
Oh yea, probably lines 9 && 14 answer my question :)